### PR TITLE
Introduce an upgrade icon

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Icon can easily be changed later if we find a xml of the mock used in Figma. I took the one named "upgrade" in the official material icon set

2022-10-15-003615_284x505_scrot

### CODE REVIEW

**Comment by Arthur-Milchior** [comment]
Added in the first message

**Review by mikehardy — APPROVED** [review]
I find the use of the word upgrade, in the context of this icon and the initial "sales pitch" to users, works, but the "upgrade" *is* a migration (your branch is even called that ;-) - just for some direction on words, unless there is consensus otherwise?

(if so, please let me know)

**Comment by david-allison** [comment]
I don't like `upgrade`, I like `speed`, and a few in:

Screenshot 2022-10-13 at 23 56 27
Screenshot 2022-10-13 at 23 56 05

**Comment by Arthur-Milchior** [comment]
> I find the use of the word upgrade, in the context of this icon and the initial "sales pitch" to users, works, but the "upgrade" _is_ a migration (your branch is even called that ;-) - just for some direction on words, unless there is consensus otherwise?
> 
> (if so, please let me know)

I've no idea what your request is. Do you want me to change a word? are you just implying that the "upgrade" word does not justify the "upgrade" icon and so we should change it ?

**Comment by Arthur-Milchior** [comment]
I updated the screenshot and code to use `speed`

**Comment by mikehardy** [comment]
Dang - I hate to bike-shed on an icon, but I like the up arrow, chrome (which has a frighteningly large percentage of all web users...) has been training users to see that and speed to me just implies performance. This is a really important thing we need users to do 

The only thing I did not really like about the original proposal was that it was so serene, and passive sitting there. I want it to be a semi-shocking call to action

update-google-chrome-red-640px

How about something in-your-face like that, red circle, fat white arrow?

Looks like "arrow-circle-up" white center with red fill is pretty close to that.

There may be better ways to do it, but here's what I'd propose, concretely


arrow_circle_up_FILL1_wght600_GRAD200_opsz48

**Comment by mikehardy** [comment]
I posted on #scoped-storage on discord my plea for single naming choice, and argued there for my choice, we can decide it there

**Comment by david-allison** [comment]
> Dang - I hate to bike-shed on an icon, but I like the up arrow, chrome (which has a frighteningly large percentage of all web users...) has been training users to see that and speed to me just implies performance. This is a really important thing we need users to do
> 
> The only thing I did not really like about the original proposal was that it was so serene, and passive sitting there. I want it to be a semi-shocking call to action
> 
> update-google-chrome-red-640px
> 
> How about something in-your-face like that, red circle, fat white arrow?
> 
> Looks like "arrow-circle-up" white center with red fill is pretty close to that.
> 
> There may be better ways to do it, but here's what I'd propose, concretely
> 
> arrow_circle_up_FILL1_wght600_GRAD200_opsz48

nice (as long as it's not red - we should use a badge for the 'urgency' when we want to).

**Comment by mikehardy** [comment]
Maybe yellow? I consider the user's need to do the storage migration urgent for the user, with risk of data loss it seems like what else could be urgent? I want to grab the user and shake them and have them do it, am I wrong in this? (maybe I'm missing something and I'm wrong...)

**Comment by Arthur-Milchior** [comment]
A new version with yellow and arrow just was added in the first message

**Comment by Arthur-Milchior** [comment]
Also, the icon text now use "migrate"

**Comment by mikehardy** [comment]
Waiting on a re-look from David since they have expressed opinions - need to see how it works for them now...

**Comment by Arthur-Milchior** [comment]
In this case let's ping @david-allison

**Comment by Arthur-Milchior** [comment]
Did a very small change. Instead of `offerToMigrate` the function is now called `startMigrationFlow`.
 That's because on the latest figma, I saw that we are actually starting the whole flow. Checking whether there was a recent backup, and offering to do so if necessary

**Review by BrayanDSO — APPROVED** [review]
FWIW, LGTM. But I believe that this is pending on David's review, as they are assigned to the issue, so I won't merge this now

File: AnkiDroid/src/main/res/values/01-core.xml [hunk 1]
```diff
@@ -156,6 +156,7 @@
     <string name="empty_cloze_warning">Edit this note and add some cloze deletions. (%1$s)</string>
     <string name="button_sync" comment="Text of the sync button">Sync</string>
     <string name="button_sync_disabled" comment="Text of the sync button when disabled">Sync temporarily disabled</string>
+    <string name="button_upgrade" comment="Text of the sync button">Migrate to scoped storage</string>
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/01-core.xml:159]: What was the final decision on the Figma? 

My belief was that this icon was for setting a backup, or are we using the same icon for 'user said no after syncing' - you need to migrate.

Comment (mikehardy) [reply AnkiDroid/src/main/res/values/01-core.xml:159]: I just read the figma, I think this should be for display on deck picker until the migration is complete.

I see there are two proposed flows (which is confusing, we must have one flow and I cannot do any PM work until the flow is agreed, really)

In either flow though, the user should see this when they have not migrated whether or not they have set a backup, in my opinion.

Comment (mikehardy) [reply AnkiDroid/src/main/res/values/01-core.xml:159]: (given that, I believe the string stands...)

**Comment by mikehardy** [comment]
contains strings + strings are clear right now (a separate effort, that I want to take advantage of) - pending merge then a string resync

**Pull request merged**